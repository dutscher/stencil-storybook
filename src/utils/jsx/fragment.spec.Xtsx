import { newSpecPage } from '@stencil/core/testing';
import { Component, h } from '@stencil/core';
import Fragment from './fragment';

describe('Fragment', () => {
  it('default', () => {
    // @ts-ignore
    expect(Fragment()).toEqual([]);
  });

  it('with props', () => {
    expect(Fragment('jojo')).toEqual([]);
  });

  it('with childs', () => {
    expect(Fragment(null, 'jojo')).toEqual(['j', 'o', 'j', 'o']);
  });

  it('test in jsx', async () => {
    @Component({
      tag: 'my-cmp',
    })
    class MyCmp {
      render() {
        return (
          <Fragment>
            <h1>Hello, World</h1>
            <p>Jojo</p>
          </Fragment>
        );
      }
    }

    const page = await newSpecPage({
      components: [MyCmp],
      html: `<my-cmp></my-cmp>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
