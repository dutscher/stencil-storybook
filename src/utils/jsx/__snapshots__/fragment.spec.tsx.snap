// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragment test in jsx 1`] = `
<my-cmp>
  <h1>
    Hello, World
  </h1>
  <p>
    Jojo
  </p>
</my-cmp>
`;
